crates/matrix-sdk-ui/src/timeline/event_item/mod.rs

```diff
@@ -350,13 +368,20 @@ impl EventTimelineItem {
     /// Gets the [`ShieldState`] which can be used to decorate messages in the
     /// recommended way.
     pub fn get_shield(&self, strict: bool) -> Option<ShieldState> {
-        self.encryption_info().map(|info| {
-            if strict {
-                info.verification_state.to_shield_state_strict()
-            } else {
-                info.verification_state.to_shield_state_lax()
+        if self.is_room_encrypted {
```

Comment:
Can we maybe bail out early here and reduce the indendation level if not encrypted?